packages/react/src/use-render/useRender.ts

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import type { ComponentRenderFn } from '../utils/types';
+import { useComponentRenderer } from '../utils/useComponentRenderer';
+import { defaultRenderFunctions } from '../utils/defaultRenderFunctions';
+import { CustomStyleHookMapping } from '../utils/getStyleHookProps';
+
+const defaultState = {};
+
+/**
+ * Returns a function that renders a Base UI component.
+ */
+function useRender<State extends Record<string, unknown>, RenderedElementType extends Element>(
+  settings: useRender.Settings<State, RenderedElementType>,
+) {
+  const { render, props, state, refs } = settings;
+  const { ref, ...extraProps } = props ?? {};
+
+  const customStyleHookMapping = React.useMemo(() => {
+    return Object.keys(state ?? {}).reduce((acc, key) => {
+      acc[key as keyof State] = () => null;
+      return acc;
+    }, {} as CustomStyleHookMapping<State>);
+  }, [state]);
```

Comment:
Wonder if `useComponentRenderer` should have an option to turn off applying `data-*` attributes to avoid runtime overhead here

Reply:
Makes sense, added optional boolean setting for this.